BlogTemplate/Models/BlogDataStore.cs

```diff
@@ -323,9 +326,58 @@ public void UpdatePost(Post newPost, Post oldPost)
             SavePost(newPost);
         }
 
-        public bool CheckSlugExists(string slug)
+        public void SaveFiles(List<IFormFile> files)
+        {           
+            foreach(var file in files)
+            {
+                if(file.Length > 0)
+                {                    
+                    using (Stream uploadedFileStream = file.OpenReadStream())
+                    {
+                        byte[] buffer = new byte[uploadedFileStream.Length];
+                        uploadedFileStream.Read(buffer, 0, buffer.Length);
+                        string name = CreateFileName(file.FileName);
+                        _fileSystem.WriteFile($"{UploadsFolder}\\{name}", buffer);
+                    }
+                }
+            }
+        }
+
+        public IEnumerable<string> GetFileNames()
+        {
+            IEnumerable<string> fileNames = _fileSystem.EnumerateFiles(UploadsFolder);
+            return fileNames;
+        }
+
+        private bool CheckFileNameExists(string fileName)
         {
-            return _fileSystem.FileExists($"{PostsFolder}\\{slug}.xml");
+            return _fileSystem.FileExists($"{UploadsFolder}\\{fileName}");
         }
+
+        private string CreateFileName(string fileName)
+        {
+            string tempName = fileName;
+            string[] elements = fileName.Split(".");

```

Comment:
Don't rely on `.Split(".")` here.  you could have files with multiple periods in the name (e.g. `foo.bar.jpg`) and you'll lose part of the name.  Instead you should use `Path.GetFileNameWithoutExtension(fileName)` and `Path.GetExtension(fileName)` to get the parts you need.

Then you could do something in the loop like `$"{filename}-{i}.{extension}", which IMO is much more readable than using the array.